truss/templates/shared/middleware/binary_header_middleware.py

```diff
@@ -0,0 +1,35 @@
+from typing import Optional, Tuple
+
+from starlette.middleware.base import BaseHTTPMiddleware
+from starlette.requests import Request
+
+
+class BinaryHeaderMiddleware(BaseHTTPMiddleware):
+    def __init__(self, app, map_input: bool = True, map_output: bool = True):
+        super().__init__(app)
+        self.map_input = map_input
+        self.map_output = map_output
+
+    @staticmethod
+    def is_binary(request: Request) -> Tuple[bool, Optional[str]]:
+        content_type_header = request.headers.get("content-type")
+        return (content_type_header == "application/octet-stream", content_type_header)
+
+    async def dispatch(self, request, call_next):
```

Comment:
let's add types here, can be fairly rudimental (ie: `Callable`)

Reply:
will do